dask_sql/mappings.py

```diff
@@ -106,10 +106,26 @@ def sql_to_python_value(sql_type: str, literal_value: Any) -> Any:
         return literal_value
 
     elif sql_type.startswith("INTERVAL"):
+        # check for finer granular interval types, e.g., INTERVAL MONTH, INTERVAL YEAR
+        try:
+            interval_type = sql_type.split()[1].lower()
+
+            if interval_type in {"year", "quarter", "month"}:
+                # if sql_type is INTERVAL YEAR, Calcite will covert to months
+                delta = pd.tseries.offsets.DateOffset(months=float(str(literal_value)))
+                return delta
+        except IndexError:
+            # no finer granular interval type specified
+            pass
+        except TypeError:
+            # interval type is not recognized, fall back to default case
+            pass
+
         # Calcite will always convert to milliseconds
         # no matter what the actual interval is
         # I am not sure if this breaks somewhere,
         # but so far it works
+        # Issue: if sql_type is INTERVAL MICROSECOND, and value <= 1000, literal_value will be rounded to 0
         return timedelta(milliseconds=float(str(literal_value)))
```

Comment:
What is the benefit of your class (`pd.tseries.offsets.DateOffset`) to the `datetime` timedelta? Just for me to understand - I am very fine with changing it

Reply:
With `pd.tseries.offsets`, my understanding is that it's more convenient to specify timedelta with data-dependent length. E.g., when incrementing the current timestamp by "1 month", with standard timedelta, we need to compute if to add 30/31/28/29 days depending the month/year of the current timestamp. 
With `pd.tseries.offsets`, we can use `pd.tseries.offsets.DateOffset(months=1), and pandas shall take the day conversion under the hood.

Similar thing happens to "year" increment as well due to leap years.

For `last_day` implementation, it is convenient to use  `pd.tseries.offsets.MonthEnd`, which shifts the time to the last day of the month.

For more generic cases (such as + hours/minutes/seconds), I think the standard timedelta and `pd.tseries.offsets` will all do the tricks.